src/test/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDIProviderRootGroupValidationTest.scala

```diff
@@ -0,0 +1,133 @@
+package mesosphere.marathon.core.externalvolume.impl.providers
+
+import com.wix.accord.{ RuleViolation, Success, Failure, Result }
+import mesosphere.marathon.api.v2.Validation
+import mesosphere.marathon.core.externalvolume.ExternalVolumes
+import mesosphere.marathon.state._
+import org.apache.mesos.{ Protos => MesosProtos }
+import org.scalatest.{ FunSuite, GivenWhenThen, Matchers }
+import play.api.libs.json.{ JsString, Json }
+
+import scala.collection.immutable.Seq
+
+class DVDIProviderRootGroupValidationTest extends FunSuite with Matchers with GivenWhenThen {
+
+  test("two volumes with different names should not result in an error") {
+    val f = new Fixture
+    Given("a root group with two apps and conflicting volumes")
+    val rootGroup = Group(
+      id = PathId.empty,
+      groups = Set(
+        Group(
+          id = PathId("/nested"),
+          apps = Set(
+            f.appWithDVDIVolume(appId = PathId("/nested/foo1"), volumeName = "vol1"),
+            f.appWithDVDIVolume(appId = PathId("/nested/foo2"), volumeName = "vol2")
+          )
+        )
+      )
+    )
+
+    f.checkResult(
+      rootGroup,
+      expectedViolations = Set.empty
+    )
+  }
+
+  test("two volumes with same name result in an error") {
+    val f = new Fixture
+    Given("a root group with two apps and conflicting volumes")
+    val app1 = f.appWithDVDIVolume(appId = PathId("/nested/app1"), volumeName = "vol")
+    val app2 = f.appWithDVDIVolume(appId = PathId("/nested/app2"), volumeName = "vol")
+    val rootGroup = Group(
+      id = PathId.empty,
+      groups = Set(
+        Group(
+          id = PathId("/nested"),
+          apps = Set(app1, app2)
+        )
+      )
+    )
+
```

Comment:
TODO: `test("two volumes with same name, one in an app where instances=0, should succeed")`